src/definition/populate_test.go

```diff
@@ -80,6 +80,76 @@ func metricsNamingManipulator(ms metric.MetricSet, entity sdk.Entity, clusterNam
 	return ms.SetMetric("displayName", entity.Name, metric.ATTRIBUTE)
 }
 
+func TestIntegrationProtocol2PopulateFunc_AddingNewMetricSetToEntity(t *testing.T) {
+	rawGroupsSample1 := RawGroups{
+		"test": {
+			"entity_id_1": RawMetrics{
+				"raw_metric_name_1": 1,
+			},
+		},
+	}
+	rawGroupsSample2 := RawGroups{
+		"test": {
+			"entity_id_1": RawMetrics{
+				"raw_metric_name_2": "foo",
+			},
+		},
+	}
+	specs := SpecGroups{
+		"test": SpecGroup{
+			TypeGenerator: fromGroupEntityTypeGuessFunc,
+			Specs: []Spec{
+				{"metric_1", FromRaw("raw_metric_name_1"), metric.GAUGE, true},
+				{"metric_2", FromRaw("raw_metric_name_2"), metric.ATTRIBUTE, true},
+			},
+		},
+	}
+	integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", new(struct{}))
```

Comment:
Why using `new` here?
```suggestion
	integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", struct{}{})
```

Reply:
I was missing old style C++ 🤣 , same c&p :shame:

Reply:
mmm interesting ,panics when initialize without new, looks like reflection is not being correctly handled in the sdk
```
--- FAIL: TestIntegrationProtocol2PopulateFunc_GeneratesSingleMetricSetForSingleEntityForMetricsFromMultipleGroups (0.00s)
panic: reflect: call of reflect.Value.Elem on struct Value [recovered]
	panic: reflect: call of reflect.Value.Elem on struct Value
```

Reply:
Maybe it must be a pointer type?

Reply:
```diff
diff --git src/definition/populate_test.go src/definition/populate_test.go
index 440d170..0d0c248 100644
--- src/definition/populate_test.go
+++ src/definition/populate_test.go
@@ -105,7 +105,7 @@ func TestIntegrationProtocol2PopulateFunc_GeneratesSingleMetricSetForSingleEntit
                        },
                },
        }
-       integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", new(struct{}))
+       integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", &struct{}{})
        if err != nil {
                t.Fatal()
        }
```

Seems to work, but perhaps we can just stay consistent with `new()` in other test cases.